### PR TITLE
Fix filtering test names in `run_standalone_tests.sh` when checking for errors

### LINKED ISSUE #19039: Filter test names in `run_standalone_tests.sh` when checking for errors
### Description & Motivation

The run_standalone_tests.sh scripts runs the tests and then checks the output log for keywords such as "error", "exception", "failed", etc. 
https://github.com/Lightning-AI/lightning/blob/dd206a20d8cbe037912c144fc3c5e5306ac0d77e/tests/tests_pytorch/run_standalone_tests.sh#L54

It includes an extra rule to avoid false positives on keywords like "xfailed" or "on_exception" (which is a hook name). However, more false-positives can occur if the name of the tests includes one of the keywords. For example, if we have a test called

```py

@RunIf(standalone=True)
def test_for_an_exception_bla_bla():
 ...
```

### Pitch

In the check here

https://github.com/Lightning-AI/lightning/blob/dd206a20d8cbe037912c144fc3c5e5306ac0d77e/tests/tests_pytorch/run_standalone_tests.sh#L54

also exclude all test names that we ran. We have access to the list since we collect all parameterizations. 

### Alternatives

_No response_

### Additional context

_No response_

cc @carmocca

### PR DESCRIPTION
**Pull request by ryan597**
## What does this PR do?

Filters test names from the `standalone_test_output.txt`.

Remove everything inside of brackets (inclusive of the brackets) with sed, then append to a txt file which is passed to grep.

Fixes #19039 

<details>
  <summary><b>Before submitting</b></summary>

- [x] Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

@awaelchli 

----
📚 Documentation preview 📚: https://pytorch-lightning--19176.org.readthedocs.build/en/19176/

### CODE REVIEW

File: tests/run_standalone_tests.sh [hunk 1]
```diff
@@ -45,7 +45,7 @@ function show_batched_output {
   if [ -f standalone_test_output.txt ]; then  # if exists
     cat standalone_test_output.txt
     # heuristic: stop if there's mentions of errors. this can prevent false negatives when only some of the ranks fail
-    if grep -iE 'error|exception|traceback|failed' standalone_test_output.txt | grep -qvE 'on_exception|xfailed'; then
+    if grep -iE 'error|exception|traceback|failed' standalone_test_output.txt | grep -vE 'on_exception|xfailed' | grep -qvF ${parametrizations_arr[@]/#/-e }; then  # https://stackoverflow.com/a/2300653
```
[the remark this answers is not in the document]
Comment (ryan597) [reply tests/run_standalone_tests.sh:48]: Oh I see the problem. When I was testing it I used the file you linked in #19039 which removes the prefixes on the `parametrizations` as the link is a different branch to the master branch.

File: tests/run_standalone_tests.sh [hunk 2]
```diff
@@ -57,6 +57,8 @@ trap show_batched_output EXIT  # show the output on exit
 
 # remove the "tests/tests_pytorch/" path suffixes
 path_prefix=$(basename "$(dirname "$(pwd)")")/$(basename "$(pwd)")"/"  # https://stackoverflow.com/a/8223345
+testnames_arr=(${parametrizations//$path_prefix/})
+
 for i in "${!parametrizations_arr[@]}"; do
   parametrization=${parametrizations_arr[$i]//$path_prefix/}
```
Comment (carmocca) [inline tests/run_standalone_tests.sh:63]: Is this line still necessary now? You already removed the prefixes in testnames_arr

File: tests/run_standalone_tests.sh [hunk 3]
```diff
@@ -41,11 +41,12 @@ parametrizations_arr=($parametrizations)
 report=''
 
 rm -f standalone_test_output.txt  # in case it exists, remove it
+rm -f testnames.txt
```
Comment (awaelchli) [inline tests/run_standalone_tests.sh:44]:
```suggestion
rm -f testnames.txt

```
Maybe this could use an empty line, just for the eyes

**Review by awaelchli — APPROVED** [review]
Looks like you did it 🎉 🚀 🥳

**Review by carmocca — APPROVED** [review]

**Pull request merged**